### PR TITLE
Dat 4116 - mercury navigation bar redesign

### PR DESCRIPTION
**Pull request by nikodamn**
Update of tests for mercury navigation bar, menu drawer and smartbanner.

Reviewers:
@dianafa @idradm

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java [hunk 1]
```diff
@@ -127,37 +128,45 @@ public Navigation openRecentWikiActivity() {
...
       return false;
     }
   }
 
-  public boolean isFooterVisible() {
+  public boolean isLogoutLinkVisible() {
     try {
-      return homeOfFandomFooter.isDisplayed();
+      return logoutLink.isDisplayed();
     } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java [hunk 2]
```diff
@@ -90,6 +79,28 @@ public void mercury_navigation_navigationOnNonEnglishWiki() {
...
+
+    topBar.openNavigation();
+    Assertion.assertTrue(navigation.isUserAvatarVisible());
+    Assertion.assertTrue(navigation.isUserProfileLinkVisible());
+    Assertion.assertTrue(navigation.isLogoutLinkVisible());
+  }
+
+  @Execute(asUser = User.ANONYMOUS)
+  @Test(groups = "mercury_navigation_navigationElementsAnonymousUser")
+  public void mercury_navigation_navigationElementsAnonymousUser() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SmartBannerTest.java [hunk 3]
```diff
@@ -24,8 +24,15 @@ private void init() {
...
-  public void mercury_smartBanner_hasFixedPositionOnTopAndCanBeClosed() {
+  @Test(groups = "mercury_smartBanner_isNotVisibleOnScrollDown")
+  public void mercury_smartBanner_isNotVisibleOnScrollDown() {
+    init();
+
+    smartBanner.scrollDown();
+  }
+
+  @Test(groups = "mercury_smartBanner_canBeClosed")
+  public void mercury_smartBanner_canBeClosed() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SmartBannerTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 4]
```diff
@@ -6,11 +6,15 @@
...
 import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.elemnt.JavascriptActions;
 import com.wikia.webdriver.common.core.helpers.Emulator;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
 import com.wikia.webdriver.elements.common.Navigate;
 import com.wikia.webdriver.elements.mercury.components.Navigation;
 import com.wikia.webdriver.elements.mercury.components.TopBar;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import org.apache.xpath.SourceTree;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MINOR Remove this unused import 'org.apache.xpath.SourceTree'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 5]
```diff
@@ -70,4 +74,28 @@ public void mecury_topbar_switchBetweenSearchAndNavigation() {
...
+  @Test(groups = "mercury_topbar_fandomBarIsVisibleOnEnglishCommunity")
+  public void mercury_topbar_fandomBarIsVisibleOnEnglishCommunity() {
+    init();
+
+    Assertion.assertTrue(topBar.isFandomBarVisible());
+  }
+
+  @Execute(onWikia = MercuryWikis.DE_WIKI)
+  @Test(groups = "mercury_topbar_fandomBarIsNotVisibleOnNonEnglishCommunity")
+  public void mercury_topbar_fandomBarIsNotVisibleOnNonEnglishCommunity() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 6]
```diff
@@ -6,11 +6,15 @@
 import com.wikia.webdriver.common.core.annotations.Execute;
 import com.wikia.webdriver.common.core.annotations.InBrowser;
 import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.elemnt.JavascriptActions;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 7]
```diff
@@ -70,4 +74,28 @@ public void mecury_topbar_switchBetweenSearchAndNavigation() {
...
+  @Execute(onWikia = MercuryWikis.DE_WIKI)
+  @Test(groups = "mercury_topbar_fandomBarIsNotVisibleOnNonEnglishCommunity")
+  public void mercury_topbar_fandomBarIsNotVisibleOnNonEnglishCommunity() {
+    init();
+
+    Assertion.assertFalse(topBar.isFandomBarVisible());
+  }
+
+  @Test(groups = "mercury_topbar_wikiaLogoRedirectsToFandomPage")
+  public void mercury_topbar_wikiaLogoRedirectsToFandomPage() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 8]
```diff
@@ -70,4 +74,28 @@ public void mecury_topbar_switchBetweenSearchAndNavigation() {
     Assertion.assertTrue(topBar.isCloseIconVisible());
     Assertion.assertTrue(topBar.isSearchIconVisible());
   }
+
+  @Test(groups = "mercury_topbar_fandomBarIsVisibleOnEnglishCommunity")
+  public void mercury_topbar_fandomBarIsVisibleOnEnglishCommunity() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 9]
```diff
@@ -6,11 +6,15 @@
 import com.wikia.webdriver.common.core.annotations.Execute;
 import com.wikia.webdriver.common.core.annotations.InBrowser;
 import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.elemnt.JavascriptActions;
 import com.wikia.webdriver.common.core.helpers.Emulator;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
 import com.wikia.webdriver.elements.common.Navigate;
 import com.wikia.webdriver.elements.mercury.components.Navigation;
 import com.wikia.webdriver.elements.mercury.components.TopBar;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java [hunk 10]
```diff
@@ -1,5 +1,6 @@
 package com.wikia.webdriver.elements.mercury.components;
 
+import com.wikia.webdriver.common.contentpatterns.MercurySubpages;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.contentpatterns.MercurySubpages'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java [hunk 11]
```diff
@@ -115,4 +125,13 @@ public boolean isCloseIconVisible() {
       return false;
     }
   }
+
+  public boolean isFandomBarVisible() {
+    try {
+      return fandomBar.isDisplayed();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues:
* MAJOR 7 major
* MINOR 4 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java (hunk 2 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SmartBannerTest.java (hunk 3 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SmartBannerTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 12]
```diff
@@ -70,4 +70,28 @@ public void mecury_topbar_switchBetweenSearchAndNavigation() {
...
+  @Test(groups = "mercury_topbar_fandomBarIsVisibleOnEnglishCommunity")
+  public void mercury_topbar_fandomBarIsVisibleOnEnglishCommunity() {
+    init();
+
+    Assertion.assertTrue(topBar.isFandomBarVisible());
+  }
+
+  @Execute(onWikia = MercuryWikis.DE_WIKI)
+  @Test(groups = "mercury_topbar_fandomBarIsNotVisibleOnNonEnglishCommunity")
+  public void mercury_topbar_fandomBarIsNotVisibleOnNonEnglishCommunity() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 13]
```diff
@@ -70,4 +70,28 @@ public void mecury_topbar_switchBetweenSearchAndNavigation() {
...
+    init();
+
+    Assertion.assertFalse(topBar.isFandomBarVisible());
+  }
+
+  @Test(groups = "mercury_topbar_wikiaLogoRedirectsToFandomPage")
+  public void mercury_topbar_wikiaLogoRedirectsToFandomPage() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java [hunk 14]
```diff
@@ -70,4 +70,28 @@ public void mecury_topbar_switchBetweenSearchAndNavigation() {
     Assertion.assertTrue(topBar.isCloseIconVisible());
     Assertion.assertTrue(topBar.isSearchIconVisible());
   }
+
+  @Test(groups = "mercury_topbar_fandomBarIsVisibleOnEnglishCommunity")
+  public void mercury_topbar_fandomBarIsVisibleOnEnglishCommunity() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java (hunk 10 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.contentpatterns.MercurySubpages'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java (hunk 11 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 8 issues:
* MAJOR 7 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 8 issues:
* MAJOR 7 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/SmartBanner.java [hunk 15]
```diff
@@ -52,7 +51,7 @@ public SmartBanner scrollDown() {
 
     PageObjectLogging.logInfo("Smart banner position is fixed at the top");
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/SmartBanner.java]: so it's fixed or not? :D

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java [hunk 16]
```diff
@@ -90,6 +79,28 @@ public void mercury_navigation_navigationOnNonEnglishWiki() {
...
+    init();
+
+    topBar.openNavigation();
+    Assertion.assertTrue(navigation.isUserAvatarVisible());
+    Assertion.assertFalse(navigation.isUserProfileLinkVisible());
+    Assertion.assertFalse(navigation.isLogoutLinkVisible());
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: what about check if `sign in | register` visible? :)

Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: good point! resolved

File: src/test/java/com/wikia/webdriver/elements/mercury/components/SmartBanner.java (hunk 15 above)
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/SmartBanner.java]: Oops.. Good catch!

File: src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java [hunk 17]
```diff
@@ -90,6 +79,29 @@ public void mercury_navigation_navigationOnNonEnglishWiki() {
...
+
+    topBar.openNavigation();
+    Assertion.assertTrue(navigation.isUserAvatarVisible());
+    Assertion.assertTrue(navigation.isUserProfileLinkVisible());
+    Assertion.assertTrue(navigation.isLogoutLinkVisible());
+  }
+
+  @Execute(asUser = User.ANONYMOUS)
+  @Test(groups = "mercury_navigation_navigationElementsAnonymousUser")
+  public void mercury_navigation_navigationElementsAnonymousUser() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SmartBannerTest.java (hunk 3 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SmartBannerTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java (hunk 12 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java (hunk 13 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java (hunk 14 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/TopBarTests.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java [hunk 18]
```diff
@@ -115,4 +124,13 @@ public boolean isCloseIconVisible() {
...
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues:
* MAJOR 7 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java [hunk 19]
```diff
@@ -71,6 +73,13 @@ public Navigation clickCloseButton() {
     return new Navigation(driver);
   }
 
+  public void clickWikiaLogo() {
+    PageObjectLogging.logInfo("Click Wikia logo");
+    wait.forElementVisible(logo);
```
Comment (rybmat) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java]: wait.forelementClickable?

**Comment by rybmat** [comment]
lg

**Pull request merged**